### PR TITLE
[PyTorch] Update moving average computation for batch norm

### PR DESCRIPTION
**Pull request by notprime**
**Update and correct moving average computation for batch norm.**
It was:
moving_mean = momentum * moving_mean + (1.0 - momentum) * mean
moving_var = momentum * moving_var + (1.0 - momentum) * var

but it should be:
moving_mean = (1.0 - momentum) * moving_mean + momentum * mean
moving_var = (1.0 - momentum) * moving_var + momentum * var

**It's the new observed value that is multiplied for momentum, while the old estimated statistic 
is multiplied by 1 - momentum** 
(also see: https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html)

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by notprime** [comment]
Check on Mxnet failed, I don't know why :/

**Comment by astonzhang** [comment]
Thanks. To be consistent, please also
* modify change the value of `momentum` from 0.9 to 0.1. (see default value in https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html)
* Update `assign_moving_average` in the tensorflow implementation.

@AnirudhDagar  is working on the CI issue. Otherwise LTGM.

**Comment by notprime** [comment]
- [x]  The `momentum` value has been modified from 0.9 to 0.1
- [x] Updated the computation for MXNet alsom, the moving average computation for Tensorflow was already correct

**Comment by notprime** [comment]
[x] Final commit pushed to the repo

**Comment by AnirudhDagar** [comment]
@astonzhang I think the failure was just a flake, since I couldn't find any specific reason behind the MxNet CI timeout on this PR. I checked out the PR locally, and ran the notebook multiple times to check for any issues. I've restarted the job, let's try again.

**Review by AnirudhDagar — COMMENTED** [review]
@astonzhang I think the failure was just a flake since I couldn't find any specific reason behind the MxNet CI timeout on this PR. I checked out the PR locally and ran the notebook multiple times to check for any issues. I've restarted the job; let's try again.

**Comment by notprime** [comment]
@AnirudhDagar @astonzhang, is there anything I should add/change/modify before the merge?

File: chapter_convolutional-modern/batch-norm.md [hunk 1]
```diff
@@ -435,8 +435,8 @@ class BatchNorm(tf.keras.layers.Layer):
         super(BatchNorm, self).build(input_shape)
 
     def assign_moving_average(self, variable, value):
-        momentum = 0.9
-        delta = variable * momentum + value * (1 - momentum)
+        momentum = 0.1
+        delta = (1 - momentum) * variable + value * momentum
```
Comment (astonzhang) [inline chapter_convolutional-modern/batch-norm.md:439]:
```suggestion
        delta = variable * (1 - momentum) + value * momentum
```

**Comment by astonzhang** [comment]
@notprime thanks. Would you like to send another PR to replace your ID with your name in our acknowledgement:
https://github.com/d2l-ai/d2l-en/blob/master/chapter_preface/index.md

**Comment by notprime** [comment]
Hi @astonzhang , PR sent!
Have a nice day!

**Comment by notprime** [comment]
@astonzhang , I was thinking that some lines should be added in the BatchNorm chapter 8.5.2.3 (Batch Normalization During Prediction), describing the momentum technique for computing estimates of mean and variance and that this technique is used because feedint the entire train dataset to compute those values could be infeasible from a memory pov. Do you agree? I can easily write some lines about it, let me know

**Comment by astonzhang** [comment]
> 

Thanks. For foundational topics we intentionally keep things simple and concise :)
